quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/NestedPropertyMappingInterceptor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.quarkus.runtime.configuration;
+
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.keycloak.quarkus.runtime.configuration.mappers.PropertyMappers;
+
+import io.smallrye.config.ConfigSourceInterceptor;
+import io.smallrye.config.ConfigSourceInterceptorContext;
+import io.smallrye.config.ConfigValue;
+import io.smallrye.config.Priorities;
+import jakarta.annotation.Priority;
+
+/**
+ * Some resolution of values that come from PropertyMappers
+ * happens at the ExpressionConfigSourceInterceptor, which is after
+ * property mapping. This interceptor appears just after the expression
+ * interceptor and will restart the context for anything not actively recursing.
```

Comment:
Maybe we could elaborate a bit more on the comment here.
```suggestion
 * interceptor and will restart the context for anything not actively recursing.
 * This is needed in case the expression contains something that requires property mapping.
```
Let me know if that makes sense.